xds/internal/xdsclient/tlscreds/bundle_test.go

```diff
@@ -0,0 +1,256 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package tlscreds
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/testdata"
+
+	testgrpc "google.golang.org/grpc/interop/grpc_testing"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+)
+
+func TestValidTlsBuilder(t *testing.T) {
+	tests := []string{
+		`{}`,
+		`{"ca_certificate_file": "foo"}`,
+		`{"certificate_file":"bar","private_key_file":"baz"}`,
+		`{"ca_certificate_file":"foo","certificate_file":"bar","private_key_file":"baz"}`,
+		`{"refresh_interval": "1s"}`,
+		`{"refresh_interval": "1s","ca_certificate_file": "foo"}`,
+		`{"refresh_interval": "1s","certificate_file":"bar","private_key_file":"baz"}`,
+		`{"refresh_interval": "1s","ca_certificate_file":"foo","certificate_file":"bar","private_key_file":"baz"}`,
+	}
+
+	for _, jd := range tests {
+		t.Run(jd, func(t *testing.T) {
```

Comment:
Not too particular here.

Passing the config as the sub-test name might make the sub-test name unreadable. Adding a name field to the table would make the test a bit more verbose. Will leave it to you.

Reply:
I did add test names here and in `TestInvalidTlsBuilder`.